tmt/steps/prepare/feature.py

```diff
@@ -0,0 +1,197 @@
+            self.info('Enable EPEL on RHEL 8')
+            command = Command()
+            if self.guest.facts.is_superuser is False:
+                command += Command('sudo')
+            command1 = command + \
+                Command('subscription-manager', 'repos', '--enable',
+                        'codeready-builder-for-rhel-8-$(arch)-rpms')
+            command2 = command + \
+                Command('dnf', '-y', 'install',
+                        'https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm')
+            self.guest.execute(command1, silent=True)
+            self.guest.execute(command2, silent=True)
+        else:
+            raise tmt.utils.PrepareError('The distro of the guest is not supported.')
+
+    def disable(self, guest: Guest, logger: tmt.log.Logger) -> None:
+        guest_distro = self.get_guest_distro(guest=guest, logger=logger)
+        if guest_distro is None:
+            raise tmt.utils.PrepareError('The distro of the guest is not supported.')
+
+        if guest_distro == 'Fedora':
+            # XXX: What to do?
+            self.info('Disable epel on Fedora, do nothing ...')
+        else:
+            pass
+
+
+_FEATURES = {
+    EPEL.KEY: EPEL,
+    CRB.KEY: CRB,
+    FIPS.KEY: FIPS
+    }
+
+
+@dataclasses.dataclass
+class PrepareFeatureData(tmt.steps.prepare.PrepareStepData):
+    epel: List[str] = field(
+        default_factory=list,
+        option=('-e', '--epel'),
+        metavar='EPEL',
+        multiple=False,
+        help='epel to be enabled.',
+        normalize=tmt.utils.normalize_string_list
+        )
+
+
+@tmt.steps.provides_method('feature')
+class PrepareFeature(tmt.steps.prepare.PreparePlugin):
+    """
+    Enable common features such as epel, crb and fips on the guest
+
+    Example config:
+
+        prepare:
+            how: feature
+            epel: enabled
+
+    ...<TBD>...
+    """
+
+    _data_class = PrepareFeatureData
+
+    def go(
+            self,
+            *,
+            guest: 'Guest',
+            environment: Optional[tmt.utils.EnvironmentType] = None,
+            logger: tmt.log.Logger) -> None:
+        """ Prepare the guests """
+        super().go(guest=guest, environment=environment, logger=logger)
+
+        # Nothing to do in dry mode
+        if self.opt('dry'):
+            return
+
+        # Enable epel/crb/fips
+        for key, value in self.data.items():
+            # PrepareFeatureData(name='default-0',
+            #                    how='feature',
+            #                    order=50,
+            #                    summary=None,
+            #                    where=[],
+            #                    epel=['enabled'])
+            if key in ['name', 'how', 'order', 'summary', 'where']:
+                continue
+
+            if key not in _FEATURES:
+                raise tmt.utils.GeneralError("Unknown key")
```

Comment:
I for one would not bother with walking all keys. This makes it harder when adding new ones, we would have to review all plugins whether they need to ignore the new keys as well.

How about iterating over `_FEATURES` instead, and checking whether a key exists in `self.data`? If there's an unknown key, it would be reported by schema validation already, by `lint` command.

Reply:
Updated. Now the code logic looks like:

```python
        # Enable or disable epel/crb/fips
        for key in _FEATURES:
            value = getattr(self.data, key, None)
            if value is None:
                continue
            feature = _FEATURES[key](parent=self, guest=guest, logger=logger)
            # ...<snip>...
```